src/models/m_rsc.erl

```diff
@@ -362,19 +363,19 @@ merge_delete(WinnerId, LoserId, Context) ->
     m_rsc_update:merge_delete(WinnerId, LoserId, Context).
 
 %% @doc Update a resource
--spec update(resource(), list(), #context{}) -> {ok, resource()} | {error, term()}.
+-spec update(resource(), props(), #context{}) -> {ok, resource()} | {error, term()}.
 update(Id, Props, Context) ->
     m_rsc_update:update(Id, Props, Context).
 
--spec update(resource(), list(), list(), #context{}) -> {ok, resource()} | {error, term()}.
+-spec update(resource(), props(), list(), #context{}) -> {ok, resource()} | {error, term()}.
 update(Id, Props, Options, Context) ->
     m_rsc_update:update(Id, Props, Options, Context).
 
 
 
 
 %% @doc Duplicate a resource.
--spec duplicate(resource(), list(), #context{}) -> {ok, NewId :: resource_id()} | {error, Reason :: string()}.
+-spec duplicate(resource(), props(), #context{}) -> {ok, NewId :: resource_id()} | {error, Reason :: string()}.
```

Comment:
Line 378 is too long: -spec duplicate(resource(), props(), #context{}) -> {ok, NewId :: resource_id()} | {error, Reason :: string()}..